packages/nextjs/components/Modal.tsx

```diff
@@ -0,0 +1,44 @@
+/**
+ * Modal component: A reusable modal component.
+ */
+type ModalProps = {
+  trigger: JSX.Element;
+  modalTitle: string;
+  modalContent: JSX.Element;
+  action?: {
+    label: string;
+    onClick: () => void;
+  };
+};
+
+const Modal: React.FC<ModalProps> = ({ trigger, modalTitle, modalContent, action }) => {
```

Comment:
Basically, it's not only modal, but also trigger. So maybe it should be named like `TriggerWithModal`? Check prop types name too if you'll change component name.

Reply:
Sounds good.

Does TriggerModal work?